### PR TITLE
[Merged by Bors] - feat(RingTheory/Nilpotent): add lemma on powers of nilpotents

### PR DESCRIPTION
**Pull request by rmhi**
add lemma saying that powers of nilpotent elements are nilpotent.

---

I've added two versions of this lemma, one of which is used in another PR that I'm working on.

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/RingTheory/Nilpotent.lean [hunk 1]
```diff
@@ -55,6 +55,18 @@ theorem IsNilpotent.neg [Ring R] (h : IsNilpotent x) : IsNilpotent (-x) := by
   rw [neg_pow, hn, mul_zero]
 #align is_nilpotent.neg IsNilpotent.neg
 
+lemma IsNilpotent.pow {n : ℕ} {S : Type*} [MonoidWithZero S] {x : S}
+    (hx : IsNilpotent x) : IsNilpotent (x ^ n.succ) := by
+  obtain ⟨N,hN⟩ := hx
```
Comment (riccardobrasca) [inline Mathlib/RingTheory/Nilpotent.lean:60]:
```suggestion
  obtain ⟨N, hN⟩ := hx
```

**Review by riccardobrasca — APPROVED** [review]
Those should work for `[SemigroupWithZero](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/GroupWithZero/Defs.html#SemigroupWithZero)`, but don't worry if there are too many results to generalize.

Thanks!

bors d+

File: Mathlib/RingTheory/Nilpotent.lean (hunk 1 above)
Comment (rmhi) [reply Mathlib/RingTheory/Nilpotent.lean:60]: Many thanks for your comments.

Comment (rmhi) [reply Mathlib/RingTheory/Nilpotent.lean:60]: Thanks for pointing out that this is true works for SemigroupWithZero. Unfortunately there doesn't seem to be an instance of Pow S Nat for S a SemigroupWithZero, and I suspect that I'd break things if I defined one. I think I'll leave it as it is.

**Pull request closed without merging**